src/test/ui/intrinsics/validity_invariants_of.rs

```diff
@@ -0,0 +1,97 @@
+// run-pass
+// revisions: disabled normal sanitized
+// [disabled]compile-flags: -Zno-validity-invariant-checks
+// [sanitized]compile-flags: -Z sanitizer=memory
```

Comment:
You will need to ignore this test based on `needs-sanitizer-*`. grep for `needs-sanitizer` in the test suite for examples of use.

Reply:
Yeah, I tried just requiring `needs-sanitizer-memory` for just the `sanitized` revision, but that doesn't seem to be allowed for a revision.

I'll split the test into two and just require the sanitizer to exist for the -Z sanitizer=memory test. Alternatively, I could try and get compiletest to allow `[sanitized]needs-sanitizer-memory`, or just put the whole test as requiring it.